### PR TITLE
sign up fuctions init

### PR DESCRIPTION
**Pull request by AyoFayomi**

### CODE REVIEW

File: frontend/components/Products/SingleProduct.jsx [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-import { Badge, Box, Button, Flex, Image, Text } from '@chakra-ui/react';
+import { Badge, Box, Button, Container, Flex, Image, Text } from '@chakra-ui/react';
 import Link from 'next/link';
 
 export default function SingleProduct({ product }) {
   const { image, name, sellingPrice, costPrice, category } = product;
 
   return (
-    <>
+    <Container>
```
Comment (rammyblog) [inline frontend/components/Products/SingleProduct.jsx]: We agreed you'd take this out right? Or it looks fine now?

Comment (rammyblog) [inline frontend/components/Products/SingleProduct.jsx]: We agreed you'd take this out right? Or it look fine now?

File: frontend/package.json [hunk 2]
```diff
@@ -22,6 +23,7 @@
     "react": "18.2.0",
     "react-dom": "18.2.0",
     "react-redux": "^8.0.2",
+    "react-toastify": "^9.0.5",
```
Comment (rammyblog) [inline frontend/package.json]: Is this necessary? I thought Chakra came with a toast
 You can uninstall it using `npm uninstall package name`

File: frontend/pages/_app.js [hunk 3]
```diff
@@ -1,15 +1,18 @@
-import { ChakraProvider, Container } from '@chakra-ui/react';
-import { Provider } from 'react-redux';
-import Navbar from '../components/Navbar';
-import getStore from '../store';
+import { ChakraProvider, Container } from "@chakra-ui/react";
+import { Provider } from "react-redux";
+import Navbar from "../components/Navbar";
+import getStore from "../store";
+import "react-toastify/dist/ReactToastify.css";
```
Comment (rammyblog) [inline frontend/pages/_app.js]: Remove this

Comment (rammyblog) [inline frontend/pages/_app.js]: Remove this

File: frontend/pages/_app.js [hunk 4]
```diff
@@ -1,15 +1,18 @@
...
 
 function MyApp({ Component, pageProps }) {
   const store = getStore(pageProps.initialState);
   return (
     <Provider store={store}>
+      <ToastContainer />
       <ChakraProvider>
         <Navbar />
-        <Container maxW="container.xl" minHeight="80vh">
+        <Container maxW="100vw" >
```
Comment (rammyblog) [inline frontend/pages/_app.js]: Restore the initial container

File: frontend/pages/register.js [hunk 5]
```diff
@@ -121,41 +145,41 @@ export default function App() {
...
 
                           return error;
                         }}
                       />
                       <FormErrorMessage>{errors?.password}</FormErrorMessage>
                     </FormControl>
-                    <Button type="submit" colorScheme="purple" width="full">
-                      REGISTER
+                    <Button type="submit" colorScheme="purple" width="full" >
+                      { pending ? "Loading..." :"REGISTER"}
```
Comment (rammyblog) [inline frontend/pages/register.js]: You can just disable the button when pending

File: frontend/store/products/actions.js [hunk 6]
```diff
@@ -1,11 +1,29 @@
...
-    const response = await fetch(
-      'https://8ba7-102-89-39-122.ngrok.io/api/products/'
-    );
+    const response = await fetch(API_URL + "/api/products/");
+    const { data } = await response.json();
+    return data;
+  }
+);
+
+export const register = createAsyncThunk(
```
Comment (rammyblog) [inline frontend/store/products/actions.js]: Move this function to it's own folder

**Review by rammyblog — CHANGES_REQUESTED** [review]
I have left some comments

File: frontend/pages/register.js (hunk 5 above)
Comment (AyoFayomi) [reply frontend/pages/register.js]: The button disables that way also

**Review by rammyblog — APPROVED** [review]

**Pull request merged**